app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -267,6 +267,22 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
 
             Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
 
+            ////////////////////////////////////////////////////////////////////////////
+            // reset increment id if needed
+            ////////////////////////////////////////////////////////////////////////////
+            /* @var Mage_Sales_Model_Order $preExistingOrder */
+            $preExistingOrder = Mage::getModel('sales/order')->loadByIncrementId($parentQuote->getReservedOrderId());
+
+            if (!$preExistingOrder->isObjectNew()) {
```

Comment:
can we add check if this preExistingOrder is not tied to Bolt transaction reference we're looking at now?

Reply:
@daisy1754 

This check is indirectly performed here:
https://github.com/BoltApp/bolt-magento1/blob/a9fe2521362b8dc3df896d5cf1063ec73d253053/app/code/community/Bolt/Boltpay/Helper/Api.php#L192-L195

And I am adding the additional check to the branch.